rust/src/openvasd/config.rs

```
@@ -655,19 +691,71 @@ impl Config {
                 config.storage.fs.key = Some(key.clone());
             }
         }
+
+        let mut scan_prefs: HashMap<String, ScanPrefValue> = HashMap::new();
+        for pref in PREFERENCES.iter() {
+            match pref.default {
+                PreferenceValue::Bool(_) => {
+                    if let Some(p) = cmds.get_one::<bool>(pref.id) {
+                        scan_prefs.insert(pref.id.to_string(), ScanPrefValue::Bool(*p));
+                    } else if let Some(ScanPrefValue::Bool(p)) =
+                        config.scanner.preferences.get(pref.id)
+                    {
+                        scan_prefs.insert(pref.id.to_string(), ScanPrefValue::Bool(*p));
+                    } else {
+                        scan_prefs.insert(pref.id.to_string(), pref.default.clone().into());
```

If I understand this logic correctly, this means we silently fail when a wrong type is given and use the default instead. This seems like a UX nightmare. At the very least we should log an `error!`. Maybe even panic?